src/compile/selection/selection.ts

```diff
@@ -364,7 +364,7 @@ export function unitName(model: Model) {
     const {facet} = facetModel;
     for (const channel of FACET_CHANNELS) {
       if (facet[channel]) {
-        name += ` + '_' + (${accessPathWithDatum(facetModel.vgField(channel), 'facet')})`;
+        name += ` + '__facet-${channel}_' + (${accessPathWithDatum(facetModel.vgField(channel), 'facet')})`;
```

Comment:
@arvind  -- not sure if `unitName` has to be a valid CSS name.  (If so, do we have a Vega expression method to escape it?)

Reply:
Not sure why we need the method exposed in Vega expression? The strings could be escaped prior to passing it in? Perhaps I'm misunderstanding?